src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
If `TypedReference` is not recorded as byref, then it seems to me like it would have already had an issue before this change - it would not have been reported properly, as there are direct calls to `ReportPointersFromValueType` that bypasses `ReportPointersFromValueTypeArg`. Are you suggesting that there is an issue with the reporting for `TypedReference` as well? If so, I'll need to test with `TypedReference` as well to see the problem before fixing it.